src/store.ts

```diff
@@ -49,8 +49,14 @@ class CookieStore {
    * Respects the `request.credentials` policy.
    */
   add(request: RequestLike, response: ResponseLike): void {
-    if (request.credentials === 'omit') {
-      return
+    try {
+      if (request.credentials === 'omit') {
+        return
+      }
+    } catch {
+      // Ignore errors thrown by `request.credentials` access, such as those in cloudflare workers
```

Comment:
Have you tried `Reflect.has(request, 'credentials')`? It may provide us the safe means to check if `"credentials"` property exists on the `request` instance.